source/objects/lander.lua

```diff
@@ -30,9 +31,67 @@ local engineSound = Assets.getSound("engine")
 -- Local functions
 -- ~~~~~~~~~~~~~~~~
 
+
+
+local function recalcDefaultMass(lander)
+	local result = 0
+	-- all the masses are stored in this table so add them up
+	for i = 1, #lander.mass do
+		result = result + lander.mass[i]
+	end
+	-- return mass of all the components + mass of fuel
+	return (result + lander.fuelCapacity)
+end
+
+
+
+local function landerHasFuelToThrust(lander, dt)
+-- returns true if the lander has enough fuel for thrust
+-- returns false if not enough fuel to thrust
+-- Note: fuel can be > 0 but still not enough to thrust
+
+	local hasThrusterUpgrade = Lander.hasUpgrade(lander, Modules.thrusters)
+	if (lander.fuel - dt) >= 0 or (hasThrusterUpgrade and (lander.fuel - (dt * 0.80)) >= 0) then
+		return true
+	else
+		return false
+	end
+end
+
+
+
+local function parachuteIsDeployed(lander)
+-- return true if lander has a parachute and it is deployed
+
+	for _, moduleItem in pairs(lander.modules) do
+		if moduleItem.id == 5 then	-- 5 = parachute	
+			if moduleItem.deployed then
+				return true
+			end
+		end
+	end
+	return false
+end
```

Comment:
This could be reduced to
```lua
local function parachuteIsDeployed(lander)
	if Lander.hasUpgrade(lander, Modules.parachute) then
		return moduleItem.deployed
	end
end
```